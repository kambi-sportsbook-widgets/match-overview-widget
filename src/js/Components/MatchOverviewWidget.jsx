import React, { Component, PropTypes } from 'react';
import { ScrolledList } from 'kambi-widget-components';
import styles from './MatchOverviewWidget.scss';
import BlendedBackground from './BlendedBackground';
import Event from './Event';
import ArrowButton from './ArrowButton';
import ItemContainer from './ItemContainer';
import TournamentLogo from './TournamentLogo';

const MatchOverviewWidget = ({ events, tournamentLogo }) => (
   <div className={styles.widget}>
      <BlendedBackground />
      <ScrolledList
         renderPrevButton={props => <ArrowButton type='left' {...props} />}
         renderNextButton={props => <ArrowButton type='right' {...props} />}
         renderItemContainer={props => <ItemContainer {...props} />}
      >
         <TournamentLogo logoClassName={tournamentLogo} />
         {events.map(event =>
            <Event
               key={event.event.id}
               event={event.event}
               liveData={event.liveData}
               outcomes={event.betOffers.length > 0 ? event.betOffers[0].outcomes : []}
            />)}
      </ScrolledList>
   </div>
);

MatchOverviewWidget.propTypes = {

   /**
    * Events to display
    */
   events: PropTypes.arrayOf(PropTypes.object).isRequired,

   /**
    * Tournament logo class name.
    */
   tournamentLogo: PropTypes.string

};

MatchOverviewWidget.defaultProps = {
   tournamentLogo: null
};

export default MatchOverviewWidget;
