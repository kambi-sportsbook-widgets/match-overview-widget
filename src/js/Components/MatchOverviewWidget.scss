:local(.widget) {
   color: white;

   position: relative;
   padding-top: 12px;
}

:local(.background) {
   position: absolute;
   top: 0;
   right: 0;
   left: 0;
   display: block;
   width: 100%;
   min-width: 100%;
   min-height: 100%;
}
