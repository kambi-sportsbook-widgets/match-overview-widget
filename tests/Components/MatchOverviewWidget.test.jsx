/* eslint-env jest */
import React, { Children } from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { mount } from 'enzyme';
import MatchOverviewWidget from '../../src/js/Components/MatchOverviewWidget';

let renderer;

const mockEvent = {
   event: {
      id: 100
   },
   liveData: {},
   betOffers: []
};

let mockMobile = false;

jest.mock('../../src/js/Services/mobile', () => () => mockMobile);

jest.useFakeTimers();

describe('MatchOverviewWidget DOM rendering', () => {

   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
      mockMobile = false;
   });

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <MatchOverviewWidget events={[]} />
      )).toMatchSnapshot();
   });

   it('renders correctly with tournament logo', () => {
      expect(renderer.render(
         <MatchOverviewWidget events={[]} tournamentLogo="test_tournament_logo" />
      )).toMatchSnapshot();
   });

   it('renders correctly with events', () => {
      expect(renderer.render(
         <MatchOverviewWidget
            events={[
               Object.assign({}, mockEvent, {event: {id: 100}}),
               Object.assign({}, mockEvent, {event: {id: 200}})
            ]} />
      )).toMatchSnapshot();
   });

   it('renders correctly with events and betoffers', () => {
      expect(renderer.render(
         <MatchOverviewWidget
            events={[
               Object.assign({}, mockEvent, {event: {id: 100}, betOffers: [{outcomes: []}]}),
               Object.assign({}, mockEvent, {event: {id: 200}, betOffers: [{outcomes: []}]})
            ]} />
      )).toMatchSnapshot();
   });

});

describe('MatchOverviewWidget behaviour', () => {

   it('mounts correctly in desktop mode', () => {
      mockMobile = false;
      const wrapper = mount(<MatchOverviewWidget events={[]} />);
      expect(wrapper.debug()).toMatchSnapshot();

   });

   it('mounts correctly in mobile mode', () => {
      mockMobile = true;

      const wrapper = mount(<MatchOverviewWidget events={[]} />);

      expect(wrapper.debug()).toMatchSnapshot();

      jest.runAllTimers();

      expect(wrapper.debug()).toMatchSnapshot();
   });

});
