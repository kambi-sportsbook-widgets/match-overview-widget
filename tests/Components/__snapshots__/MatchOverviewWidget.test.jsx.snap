// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchOverviewWidget DOM rendering renders correctly with default props 1`] = `
<div
  className="widget"
>
  <BlendedBackground />
  <ScrolledList
    alignItems="center"
    renderItemContainer={[Function]}
    renderNextButton={[Function]}
    renderPrevButton={[Function]}
    scrollToItemMode="to-left"
    selected={0}
    showControls={true}
    step={2}
  >
    <TournamentLogo
      logoName={null}
    />
  </ScrolledList>
</div>
`;

exports[`MatchOverviewWidget DOM rendering renders correctly with events 1`] = `
<div
  className="widget"
>
  <BlendedBackground />
  <ScrolledList
    alignItems="center"
    renderItemContainer={[Function]}
    renderNextButton={[Function]}
    renderPrevButton={[Function]}
    scrollToItemMode="to-left"
    selected={0}
    showControls={true}
    step={2}
  >
    <TournamentLogo
      logoName={null}
    />
    <Event
      event={
        Object {
          "id": 100,
        }
      }
      liveData={Object {}}
      outcomes={Array []}
    />
    <Event
      event={
        Object {
          "id": 200,
        }
      }
      liveData={Object {}}
      outcomes={Array []}
    />
  </ScrolledList>
</div>
`;

exports[`MatchOverviewWidget DOM rendering renders correctly with events and betoffers 1`] = `
<div
  className="widget"
>
  <BlendedBackground />
  <ScrolledList
    alignItems="center"
    renderItemContainer={[Function]}
    renderNextButton={[Function]}
    renderPrevButton={[Function]}
    scrollToItemMode="to-left"
    selected={0}
    showControls={true}
    step={2}
  >
    <TournamentLogo
      logoName={null}
    />
    <Event
      event={
        Object {
          "id": 100,
        }
      }
      liveData={Object {}}
      outcomes={Array []}
    />
    <Event
      event={
        Object {
          "id": 200,
        }
      }
      liveData={Object {}}
      outcomes={Array []}
    />
  </ScrolledList>
</div>
`;

exports[`MatchOverviewWidget DOM rendering renders correctly with tournament logo 1`] = `
<div
  className="widget"
>
  <BlendedBackground />
  <ScrolledList
    alignItems="center"
    renderItemContainer={[Function]}
    renderNextButton={[Function]}
    renderPrevButton={[Function]}
    scrollToItemMode="to-left"
    selected={0}
    showControls={true}
    step={2}
  >
    <TournamentLogo
      logoName="test_tournament_logo"
    />
  </ScrolledList>
</div>
`;

exports[`MatchOverviewWidget behaviour mounts correctly in desktop mode 1`] = `
"<MatchOverviewWidget events={{...}} tournamentLogo={{...}}>
  <div className=\\"widget\\">
    <BlendedBackground>
      <div className=\\"backgroundContainer\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\" className=\\"background\\">
          <defs>
            <filter id=\\"filter\\">
              <feImage result=\\"slide2\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" preserveAspectRatio=\\"xMidYMid slice\\" xlinkHref=\\"assets/overview-bw-bg-desktop.jpg\\" />
              <feBlend in2=\\"SourceGraphic\\" in=\\"slide2\\" mode=\\"multiply\\" />
            </filter>
          </defs>
          <rect className=\\"KambiWidget-primary-color blendRect\\" x=\\"0\\" y=\\"0\\" filter=\\"url(#filter)\\" width=\\"100%\\" height=\\"100%\\" />
        </svg>
      </div>
    </BlendedBackground>
    <ScrolledList renderPrevButton={[Function]} renderNextButton={[Function]} renderItemContainer={[Function]} selected={0} scrollToItemMode=\\"to-left\\" showControls={true} step={2} alignItems=\\"center\\">
      <div className=\\"container\\" style={{...}}>
        <div className=\\"eyeshot\\">
          <div className=\\"bar\\">
            <ItemContainer selected={true} onClick={[Function]} onWidth={[Function]}>
              <div className=\\"item\\">
                <TournamentLogo logoName={{...}}>
                  <div className=\\"general\\">
                    <i className=\\"kw-custom-logo-large-type\\" style={{...}} />
                  </div>
                </TournamentLogo>
              </div>
            </ItemContainer>
          </div>
        </div>
        <ArrowButton type=\\"left\\" onClick={[Function]} disabled={true}>
          <button className=\\"left\\" onClick={[Function]} disabled={true}>
            <svg className=\\"icon\\" width=\\"22px\\" height=\\"36px\\" viewBox=\\"0 860 420 500\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\">
              <path id=\\"arrowLeft\\" transform=\\"scale(1, -1) translate(0, -1500)\\" d=\\"M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z\\" />
            </svg>
          </button>
        </ArrowButton>
        <ArrowButton type=\\"right\\" onClick={[Function]} disabled={true}>
          <button className=\\"right\\" onClick={[Function]} disabled={true}>
            <svg className=\\"icon\\" width=\\"22px\\" height=\\"36px\\" viewBox=\\"0 860 420 500\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\">
              <path id=\\"arrowLeft\\" transform=\\"scale(1, -1) translate(0, -1500)\\" d=\\"M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z\\" />
            </svg>
          </button>
        </ArrowButton>
      </div>
    </ScrolledList>
  </div>
</MatchOverviewWidget>"
`;

exports[`MatchOverviewWidget behaviour mounts correctly in mobile mode 1`] = `
"<MatchOverviewWidget events={{...}} tournamentLogo={{...}}>
  <div className=\\"widget\\">
    <BlendedBackground>
      <div className=\\"backgroundContainer\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\" className=\\"background\\">
          <defs>
            <filter id=\\"filter\\">
              <feImage result=\\"slide2\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" preserveAspectRatio=\\"xMidYMid slice\\" xlinkHref=\\"assets/overview-bw-bg-mobile.jpg\\" />
              <feBlend in2=\\"SourceGraphic\\" in=\\"slide2\\" mode=\\"multiply\\" />
            </filter>
          </defs>
          <rect className=\\"KambiWidget-primary-color blendRect\\" x=\\"0\\" y=\\"0\\" filter=\\"url(#filter)\\" width=\\"100%\\" height=\\"100%\\" />
        </svg>
      </div>
    </BlendedBackground>
    <ScrolledList renderPrevButton={[Function]} renderNextButton={[Function]} renderItemContainer={[Function]} selected={0} scrollToItemMode=\\"to-left\\" showControls={false} step={2} alignItems=\\"center\\">
      <div className=\\"container no-controls\\" style={{...}}>
        <div className=\\"eyeshot\\">
          <div className=\\"bar\\">
            <ItemContainer selected={true} onClick={[Function]} onWidth={[Function]}>
              <div className=\\"item\\">
                <TournamentLogo logoName={{...}}>
                  <div className=\\"general\\">
                    <i className=\\"kw-custom-logo-large-type\\" style={{...}} />
                  </div>
                </TournamentLogo>
              </div>
            </ItemContainer>
          </div>
        </div>
      </div>
    </ScrolledList>
  </div>
</MatchOverviewWidget>"
`;

exports[`MatchOverviewWidget behaviour mounts correctly in mobile mode 2`] = `
"<MatchOverviewWidget events={{...}} tournamentLogo={{...}}>
  <div className=\\"widget\\">
    <BlendedBackground>
      <div className=\\"backgroundContainer\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\" className=\\"background\\">
          <defs>
            <filter id=\\"filter\\">
              <feImage result=\\"slide2\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" preserveAspectRatio=\\"xMidYMid slice\\" xlinkHref=\\"assets/overview-bw-bg-mobile.jpg\\" />
              <feBlend in2=\\"SourceGraphic\\" in=\\"slide2\\" mode=\\"multiply\\" />
            </filter>
          </defs>
          <rect className=\\"KambiWidget-primary-color blendRect\\" x=\\"0\\" y=\\"0\\" filter=\\"url(#filter)\\" width=\\"100%\\" height=\\"100%\\" />
        </svg>
      </div>
    </BlendedBackground>
    <ScrolledList renderPrevButton={[Function]} renderNextButton={[Function]} renderItemContainer={[Function]} selected={1} scrollToItemMode=\\"to-left\\" showControls={false} step={2} alignItems=\\"center\\">
      <div className=\\"container no-controls\\" style={{...}}>
        <div className=\\"eyeshot\\">
          <div className=\\"bar\\">
            <ItemContainer selected={true} onClick={[Function]} onWidth={[Function]}>
              <div className=\\"item\\">
                <TournamentLogo logoName={{...}}>
                  <div className=\\"general\\">
                    <i className=\\"kw-custom-logo-large-type\\" style={{...}} />
                  </div>
                </TournamentLogo>
              </div>
            </ItemContainer>
          </div>
        </div>
      </div>
    </ScrolledList>
  </div>
</MatchOverviewWidget>"
`;
